app/controllers/concerns/caching_headers.rb

```diff
@@ -1,4 +1,4 @@
-module FastlyHeaders
+module CachingHeaders
```

Comment:
**thought (non-blocking):** I know this is outside the scope of this PR, I just want to plant the seed in more heads :wink: I still believe we could clean this up a lot by it not being a concern:

1. Make this a normal class with a single public interface method (e.g. `SetCachingHeaders.call`)
2. Add a wrapper method in `ApplicationController` so we don't have to change all the current controllers:
    ```ruby
    def set_cache_control_headers
      SetCachingHeaders.call
    end
    ```
3. Have (nested) subclasses that implement only the logic they care about, e.g. `SetCachingHeaders::Fastly` and `SetCachingHeaders::Nginx`.
4. Decide which caching strategy to use in the public method:
    ```ruby
     def self.call(**kwargs)
       @cache_klass ||= fastly_enabled? ? Fastly : Nginx
       @cache_klass.set_cache_control_headers(**kwargs)
    end
    ```

Reply:
@citizen428 These are good points, although we discovered today that we actually might not even _need_ to have 2 different code paths here at all! I've updated the PR description to reflect that (and to explain what this PR now actually does)

Reply:
Avoiding branching completely is even better if possible, thanks for the detailed explanation @vaidehijoshi!